source/Halibut/Diagnostics/CachingLogFactory.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Concurrent;
+using Halibut.Diagnostics.LogCreators;
+
+namespace Halibut.Diagnostics
+{
+    public class CachingLogFactory : ILogFactory
```

Comment:
It's not really `Caching` though is it in the way you would think of caching

Reply:
hmm do you have better name in mind?

`RetainingLogFactory`
`MemoryLeakingLogFactory`
`ThisIsWhyYourMemoryDumpIsMassiveLogFactory`